src/runtime/hexagon/hexagon/hexagon_device_api_v2.cc

```diff
@@ -119,6 +119,13 @@ void HexagonDeviceAPIv2::FreeWorkspace(Device dev, void* data) {
   workspace_allocations_.erase(it);
 }
 
+void* HexagonDeviceAPIv2::AllocVtcmWorkspace(Device dev, int ndim, const int64_t* shape,
+                                             DLDataType dtype, Optional<String> mem_scope) {
+  return AllocDataSpace(dev, ndim, shape, dtype, mem_scope);
```

Comment:
* I think we should add an assert here that `ndim` is either 1 or 2, the supported number of physical dimensions.  

* Does this call `AllocDataSpace` as implemented by `DeviceAPI`, or as implemented by `HexagonDeviceAPIV2`?  I only see an override of `HexagonDeviceAPIv2::AllocDataSpace` for the overload without a `mem_scope` argument.

Reply:
> I think we should add an assert here that ndim is either 1 or 2, the supported number of physical dimensions.

Done.  Added a check for `ndim` == 1 as follows:

```
  // Forcing contiguous allocation, for now
  // TODO(Straw): Enable discontiguous allocation
  CHECK_EQ(ndim, 1);
```

The Hexagon Device API is forcing / assuming contiguous (1d) memory allocations in all cases including for this PR.  The 6 places in the Hexagon Device API where 1d memory allocations are forced / assumed are called out in this PR.  The original comment and `TODO` items called out RFC 39 which has now landed along with PR #9727.  I have edited the `TODO` items accordingly.  These `TODO` items should be handled in a follow-up PR once all the pieces are in place for discontiguous allocation.

> Does this call AllocDataSpace as implemented by DeviceAPI, or as implemented by HexagonDeviceAPIV2? I only see an override of HexagonDeviceAPIv2::AllocDataSpace for the overload without a mem_scope argument.

It calls HexagonDeviceAPIv2::AllocDataSpace

Reply:
Thank you.  I have difficulty remembering how C++ handles function overloads of virtual functions, if a subclass only overrides some of the overloads, and need to check whenever I run into it.